### PR TITLE
add LendingPage v2

### PR DESCRIPTION
**Pull request by rick23p**

### CODE REVIEW

**Comment by rick23p** [comment]
closes #1078

**Comment by bitcoinuser** [comment]
Hi, could you please add strings to Deposit and Withdraw buttons like this? https://github.com/DistributedCollective/Sovryn-frontend/pull/1229

Thanks

File: src/app/pages/LendingPage/components/AccountBalance/index.tsx
Comment (creed-victor) [inline src/app/pages/LendingPage/components/AccountBalance/index.tsx:66]: Repay is from borrow (loans), everything that is related to borrows should removed / dropped from lending page.

File: src/app/pages/LendingPage/components/CurrencyContainer/PoolChart.tsx
Comment (creed-victor) [inline src/app/pages/LendingPage/components/CurrencyContainer/PoolChart.tsx:12]: There is APR data in the backend (old lending) which can be used to show at least one of the lines i think.

File: src/app/pages/LendingPage/components/HistoryTable/index.tsx
Comment (creed-victor) [inline src/app/pages/LendingPage/components/HistoryTable/index.tsx:20]: History is already available in the backend (see old lending history component)

File: src/app/pages/LendingPage/types.ts
Comment (creed-victor) [inline src/app/pages/LendingPage/types.ts:5]:
```suggestion
export interface LendingPageState {
```

File: src/app/pages/LendingPage/types.ts
Comment (creed-victor) [inline src/app/pages/LendingPage/types.ts:24]: Not related to lending

**Review by creed-victor — CHANGES_REQUESTED** [review]

**Comment by bitcoinuser** [comment]
When the user didn't lend yet, I think should have a message like: (Lend your asset and earn income). 
Something like what happens on yield farming and what I proposed here: https://github.com/DistributedCollective/Sovryn-frontend/pull/1230

Captura de tela de 2021-05-31 15-10-34

File: src/app/pages/LendingPage/Loadable.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/Loadable.tsx:3]: I don't think this comment is up-to-date (and I also don't think we need it).

File: src/app/pages/LendingPage/assets/index.scss
Comment (tiltom) [inline src/app/pages/LendingPage/assets/index.scss:1]: Why do you use `scss`? We should use `Tailwind` as much as possible (and I see several definitions from this file where it's possible) or `styled components` for the rest.

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:60]: Unnecessary casting.

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:62]: Unnecessary casting.

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:68]: Unnecessary casting.

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:71]:
```suggestion
  const isGreaterThanZero = useMemo(() => bignumber(weiAmount).greaterThan(0), [weiAmount]);
```

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:110]: Rename to `isValid`.

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:117]: Rename to `isValidRedeem`.

**Comment by bitcoinuser** [comment]
> When the user didn't lend yet, I think should have a message like: (Lend your asset and earn income).
> Something like what happens on yield farming and what I proposed here: #1230
> 
> Captura de tela de 2021-05-31 15-10-34

Hi, I see you added (Add liquidity and earn income) instead of (Lend your asset and earn income). But I still think should be (Lend your asset and earn income). 
Why? Add liquidity it's confusing because looks like something related to Yield Farm and not Lending. I hope you reconsider it, thanks.

**Comment by bitcoinuser** [comment]
Hi,

The B symbol is being showed on asset other than rBTC: 

Captura de tela de 2021-06-01 10-24-34

**Comment by bitcoinuser** [comment]
Months are in english language. I would like to see in PT-BR when sovryn is in PT-BR.

Captura de tela de 2021-06-01 10-28-05

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:140]: Rewrite it similarily to `txFeeArgs` and rename to `getMethodName`.

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:255]: This logic shouldn't be here. Create a method `handleSubmit` that will decide what handler it wants to call.

**Comment by bitcoinuser** [comment]
Hi, Please add these missing strings to be able to translate to other languages:

Captura de tela de 2021-06-01 10-31-59

File: src/app/pages/LendingPage/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/index.tsx:14]: Unnecessary, you don't need an empty props type.

File: src/app/pages/LiquidityMining/components/AddLiquidityDialog/AddLiquidityDialogV1.tsx
Comment (tiltom) [inline src/app/pages/LiquidityMining/components/AddLiquidityDialog/AddLiquidityDialogV1.tsx:71]: Rename to `hasSufficientBalance` and you don't need `{}` and `return`.

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:75]: Rename to `hasSufficientBalance`.

File: src/locales/en/translation.json
Comment (tiltom) [inline src/locales/en/translation.json:793]: Make sure you have final texts.

**Review by tiltom — CHANGES_REQUESTED** [review]

**Comment by tiltom** [comment]
Also, the page isnt' responsive at all between 1200px and 1500px. Here is for example 1390px width.

image

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:837]:
```suggestion
        "apy": "APY <0/> variable"
```

Comment (bitcoinuser) [reply src/locales/en/translation.json:837]: Already have %. 

Captura de tela de 2021-06-01 11-17-01

File: src/app/components/FinanceV2Components/LootDrop/index.tsx
Comment (rick23p) [reply src/app/components/FinanceV2Components/LootDrop/index.tsx:8]: Well on the design it's only one asset:

Screen Shot 1400-03-11 at 7 23 46 PM

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:830]:
```suggestion
        "apy": "APY",
```

Comment (bitcoinuser) [reply src/locales/en/translation.json:830]: No need % in this case, as the result already have %

Captura de tela de 2021-06-01 14-05-50

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:797]:
```suggestion
    "deposit": "Lend",
```

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:798]:
```suggestion
    "withdraw": "Redeem",
```

File: src/locales/en/translation.json
Comment (bitcoinuser) [reply src/locales/en/translation.json:797]: In my option make sense to rename the button to Lend because when clicked will have a popup with  button to Deposit.

File: src/locales/en/translation.json
Comment (bitcoinuser) [reply src/locales/en/translation.json:798]: I think when you lend money in a system, then you redeem it.

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:828]: In my opinion, the (asset) word should be replaced by the asset name. rBTC, DoC....

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:844]:
```suggestion
        "title": "Redeem",
```

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:845]:
```suggestion
        "cta": "Redeem",
```

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:832]:
```suggestion
        "myProfit": "Yield",
```

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:838]: Lend

Comment (bitcoinuser) [inline src/locales/en/translation.json:838]:
```suggestion
        "title": "Lend",
```

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:832]:
```suggestion
        "myProfit": "Profit",
```

File: src/app/components/FinanceV2Components/LootDrop/index.tsx
Comment (tiltom) [reply src/app/components/FinanceV2Components/LootDrop/index.tsx:8]: Also, that loot drop is obsolete (look at the date) and we don't even have ETH Lend. So ask @t-z-u what do we want here

File: src/locales/en/translation.json
Comment (tiltom) [reply src/locales/en/translation.json:793]: I still don't think those are the final texts, ask @t-z-u .

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:156]: Just use a ternary expression and rewrite it to `const handleSubmit = () =>  type === 'add' ? handleLendSubmit() : handleUnlendSubmit();` .

Also, use `useCallback`.

File: src/app/pages/LendingPage/components/LendingDialog/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/LendingDialog/index.tsx:141]: Use `useCallback` here and please, don't write conditions and return on one line (if it's not a ternary expression).

Just add `{}` and some spacing.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/components/FinanceV2Components/PieChart/index.tsx
Comment (tiltom) [inline src/app/components/FinanceV2Components/PieChart/index.tsx:32]: Don't use nested ternary expressions, create a method that will return the color and use it here.

File: src/app/components/FinanceV2Components/utils/getAssetColor.ts
Comment (tiltom) [inline src/app/components/FinanceV2Components/utils/getAssetColor.ts:21]: I am not sure about this naming, rename it to `getAssetSecondaryColor` or something similar.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/locales/en/translation.json
Comment (peymannaderi3) [reply src/locales/en/translation.json:820]: I just talked to tzu and he said it's the final message.

**Review by tiltom — APPROVED** [review]

**Comment by bitcoinuser** [comment]
Hi, yesterday didn't have this pizza chart and was great. Today I see this and it's ugly:

Captura de tela de 2021-06-02 10-30-53

File: src/app/pages/LendingPage/components/CurrencyContainer/CurrencyRow.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/CurrencyContainer/CurrencyRow.tsx:50]: Well, this doesn't work as intended. This is what I see on 1244px and it shouldn't be there.

image

Also, it's not a good idea to read screen width from window, just create a styled component and use a media query in it (with `display: none`).

**Review by tiltom — CHANGES_REQUESTED** [review]

**Comment by bitcoinuser** [comment]
> Hi,
> 
> The B symbol is being showed on asset other than rBTC:
> 
> Captura de tela de 2021-06-01 10-24-34

Look this too:  The B symbol on DOC chart.

Captura de tela de 2021-06-02 10-30-53

**Comment by bitcoinuser** [comment]
I don't see why this info is necessary to be showed on lend page (will only confuse users). This info already exist on Stats page. 

Captura de tela de 2021-06-02 10-49-34

**Comment by bitcoinuser** [comment]
Hi, I think when the user has 0 balance on specific asset, the button Deposit should be inactive too. And the message (Lend your BPRO and earn income) should be (You don't have BPRO on your wallet).
Obs. Maybe in the future, in these cases, the button Deposit could be a Buy button.

Captura de tela de 2021-06-02 21-48-03

**Review by tiltom — CHANGES_REQUESTED** [review]
I noticed two things on the testnet:

1. The Pie chart colors are switched (compare the charts with numbers).
2. If you haven't lent anything, a table row will weirdly collapse under 1200px width.

image

**Review by tiltom — DISMISSED** [review]

**Review by creed-victor — APPROVED** [review]
Approving feature, but do not merge this until borrowing is done too (both features must be released in the same time)

I'm also with bitcoinuser regarding Deposit and Withdraw renaming to Lend and Redeem (actually it's not even renaming but keeping naming as it's in current version).

Also, I'm not fan of having pie charts here, especially that chart colors are synced with asset colors.
In ideal way we want to make page once and implement new pool in 5 minutes - one person adds asset name and smart-contract address to the pool dictionary and just deploys it. Having custom pies means that dev now also needs a color code from designer and get approval if it looks good.
It also takes a lot of space while being not very important thing, in my opinion users who opens lending page for the first time will have hard time figuring out what this pie shows - ordering is not really self-explanatory (chart and legend is link is detached by pool name in the middle)

File: src/app/pages/LendingPage/components/HistoryTable/TableRow/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/HistoryTable/TableRow/index.tsx:37]: I would use `AssetSymbolRenderer`.

File: src/app/pages/LendingPage/components/HistoryTable/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/HistoryTable/index.tsx:57]: I would use `useBlockSync` hook and add its result to the dependencies array so the table is refreshed automatically.

File: src/app/pages/LendingPage/components/HistoryTable/index.tsx
Comment (tiltom) [inline src/app/pages/LendingPage/components/HistoryTable/index.tsx:59]: Use `useCallback`.

**Review by tiltom — CHANGES_REQUESTED** [review]

**Comment by bitcoinuser** [comment]
Hi, on mine is showing a horizontal scroll.

Captura de tela de 2021-06-29 11-37-37

File: src/locales/en/translation.json
Comment (bitcoinuser) [inline src/locales/en/translation.json:1080]:
```suggestion
      "apy": "{{asset}} APY",
```

**Review by tiltom — APPROVED** [review]

**Review by creed-victor — APPROVED** [review]

**Pull request merged**
